compiler/sem/semexprs.nim

```diff
@@ -3491,10 +3491,22 @@ proc hoistParamsUsedInDefault(c: PContext, call, letSection, defExpr: var PNode)
         newNodeI(nkEmpty, letSection.info),
         call[paramPos])
 
-      call[paramPos] = newSymNode(hoistedVarSym) # Refer the original arg to its hoisted sym
+      if hoistedVarSym.typ.kind == tyVar:
```

Comment:
we don't need to consider any skipTypes/unwrapping because this is after generic resolution etc, yes?

Reply:
In part, yes. There's also the assumption throughout parts of the compiler that `tyVar` cannot appear be nested in a `tyAlias` or the like.